astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -163,46 +305,60 @@ class Galactocentric(BaseCoordinateFrame):
-
-        # backwards-compatibility
-        if ('galcen_ra' in kwargs or 'galcen_dec' in kwargs):
-            warnings.warn("The arguments 'galcen_ra', and 'galcen_dec' are "
-                          "deprecated in favor of specifying the sky coordinate"
-                          " as a CoordinateAttribute using the 'galcen_coord' "
-                          "argument", AstropyDeprecationWarning)
-
-            galcen_kw = dict()
-            galcen_kw['ra'] = kwargs.pop('galcen_ra', self.galcen_coord.ra)
-            galcen_kw['dec'] = kwargs.pop('galcen_dec', self.galcen_coord.dec)
-            kwargs['galcen_coord'] = ICRS(**galcen_kw)
+        # Set default frame attribute values based on the ScienceState instance
+        # for the solar parameters defined above
+        default_params = galactocentric_frame_defaults.get()
+        self.frame_attribute_references = \
+            galactocentric_frame_defaults._references.copy()
+
+        warn = False
+        for k in default_params:
+            # If a frame attribute is set by the user, remove its reference
+            self.frame_attribute_references.pop(k, None)
+
+            # If a parameter is read from the defaults, we might want to warn
+            # the user that the defaults will change (see below)
+            if k not in kwargs:
+                warn = True
+
+            # Keep the frame attribute if it is set by the user, otherwise use
+            # the default value
+            kwargs[k] = kwargs.get(k, default_params[k])
+
+        # If the frame defaults have not been updated with the ScienceState
+        # class, and the user uses any default parameter value, raise a
+        # deprecation warning to inform them that the defaults will change in
+        # the future:
+        if galactocentric_frame_defaults._value == 'pre-v4.0' and warn:
+            docs_link = 'http://docs.astropy.org/en/latest/coordinates/galactocentric.html'
+            warnings.warn('In v4.1 and later versions, the Galactocentric '
+                          'frame will adopt default parameters that may update '
+                          'with time. An updated default parameter set is '
+                          'already available through the '
+                          'astropy.coordinates.galactocentric_frame_defaults '
+                          'ScienceState object, as described in but the '
+                          'default is currently still set to the pre-v4.0 '
+                          'parameter defaults. The safest way to guard against '
+                          'changing default parameters in the future is to '
+                          'either (1) specify all Galactocentric frame '
+                          'attributes explicitly when using the frame, '
+                          'or (2) set the galactocentric_frame_defaults '
+                          'parameter set name explicitly. See {} for more '
+                          'information.'.format(docs_link),
```

Comment:
You should use an f-string.
```suggestion
                          'information.',
```